test/controllers/accounts/terms_controller_test.rb

```diff
@@ -48,32 +48,72 @@ def test_show_not_seen_with_referer
       assert_response :success
     end
 
-    def test_update_not_seen_without_referer
-      user = create(:user, :terms_seen => false, :terms_agreed => nil)
+    def test_update_decline
+      user = create(:user, :terms_seen => false, :terms_agreed => nil, :tou_agreed => nil)
+      session_for(user)
+
+      freeze_time do
+        put account_terms_path
+
+        assert_redirected_to account_path
+        assert_equal({ :partial => "accounts/terms/terms_declined_flash" }, flash[:notice])
+
+        user.reload
+        assert user.terms_seen
+        assert_nil user.terms_agreed
+        assert_nil user.tou_agreed
+      end
+    end
+
+    def test_update_not_seen
+      user = create(:user, :terms_seen => false, :terms_agreed => nil, :tou_agreed => nil)
       session_for(user)
 
-      put account_terms_path, :params => { :read_ct => 1, :read_tou => 1 }
-      assert_redirected_to account_path
-      assert_equal "Thanks for accepting the new contributor terms!", flash[:notice]
+      freeze_time do
+        put account_terms_path, :params => { :read_ct => 1, :read_tou => 1 }
 
-      user.reload
+        assert_redirected_to account_path
+        assert_equal "Thanks for accepting the new contributor terms!", flash[:notice]
 
-      assert_not_nil user.terms_agreed
-      assert user.terms_seen
+        user.reload
+        assert user.terms_seen
+        assert_equal Time.now.utc, user.terms_agreed
+        assert_equal Time.now.utc, user.tou_agreed
+      end
     end
 
     def test_update_not_seen_with_referer
-      user = create(:user, :terms_seen => false, :terms_agreed => nil)
+      user = create(:user, :terms_seen => false, :terms_agreed => nil, :tou_agreed => nil)
+      session_for(user)
+
+      freeze_time do
+        put account_terms_path, :params => { :referer => "/test", :read_ct => 1, :read_tou => 1 }
+
+        assert_redirected_to "/test"
+        assert_equal "Thanks for accepting the new contributor terms!", flash[:notice]
+
+        user.reload
+        assert user.terms_seen
+        assert_equal Time.now.utc, user.terms_agreed
+        assert_equal Time.now.utc, user.tou_agreed
+      end
+    end
+
+    def test_update_previously_accepted
```

Comment:
We should probably have `test_decline_previously_accepted` to check that the dates don't change if you decline when revisiting the terms.